### ISSUE TITLE
Authentication pod timeout with Google

### ISSUE THREAD

**Issue by ReneElstgeest** [pomerium/pomerium#572]
First of all: thanks for creating this. Up until now I was very happy with your solution :-) (and I think after this I still will be)

## What happened?
Got a 503 error on all ingresses redirecting to Pomerium proxy.

## What did you expect to happen?
To get the log in screen of my Google app.

## How'd it happen?

1. Ran `Opened my service where I needed authentication to get access.`
2. Clicked `Nothin`
3. Saw error `503`

## What's your environment like?

- Pomerium version latest helm chart from `pomerium        https://helm.pomerium.io`
- Server Operating System/Architecture/Cloud: Kubernetes on Ubuntu

## What's your config.yaml?

```config.yaml
config:
  # routes under this wildcard domain are handled by pomerium
  rootDomain: elstgeest.com
  existingConfig: "pomerium"

authenticate:
  # see https://www.pomerium.io/docs/identity-providers.html
  idp:
    provider: google
    clientID: "the client id"
    clientSecret: "value of the secret"

ingress:
  secretName: "letsencrypt-production"
```

## What did you see in the logs?

```logs
{"level":"info","version":"v0.6.2+5f67a70","time":"2020-03-28T07:34:29Z","message":"cmd/pomerium"}
{"level":"debug","time":"2020-03-28T07:34:29Z","message":"internal/grpc: added custom certificate authority"}
{"level":"fatal","error":"Get https://accounts.google.com/.well-known/openid-configuration: dial tcp: lookup accounts.google.com on 10.152.183.10:53: read udp 10.1.100.46:36066->10.152.183.10:53: read: connection refused","time":"2020-03-28T07:34:49Z","message":"cmd/pomerium"}
```

## Additional context

none, for now.

Thnx in advance for helping out.

**Comment by desimone** [comment]
@ReneElstgeest looks like a possible DNS issue.

`dial tcp: lookup accounts.google.com on 10.152.183.10:53: read udp 10.1.100.46:36066->10.152.183.10:53: read: connection refused `

Your DNS server `10.152.183.10:53` is not returning a result for gsuite.

**Comment by ReneElstgeest** [comment]
Ok, I checked, but the cluster is well able to get `https://accounts.google.com/.well-known/openid-configuration` from Google. So DNS on the server is working fine.
The pod is currently crashing before I can get into `/bin/bash` to see whether it can see the configuration. So, I'm trying to figure out how to do that.

**Comment by ReneElstgeest** [comment]
Ok, I'm stuck now. Either I am to early and I get this:
`Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "b83824b832eb16364a6717503f8aae1dc80b3f06f2eed4bb45457bf1e243f23a": OCI runtime exec failed: exec failed: container_linux.go:345: starting container process caused "exec: \"/bin/bash\": stat /bin/bash: no such file or directory": unknown`
Or I get:
`unable to upgrade connection: container not found ("pomerium")`

**Comment by desimone** [comment]
> The pod is currently crashing before I can get into /bin/bash to see whether it can see the configuration. So, I'm trying to figure out how to do that.

Our docker image is [distroless](https://github.com/GoogleContainerTools/distroless) which is great for size, and security but lacks a shell. That said, we should have a "debug" image available to make testing easier. I will follow up with an issue and report back. 

I am almost positive this is a DNS issue based of the error message, however, I'm not sure of the source.

**Comment by ReneElstgeest** [comment]
> I am almost positive this is a DNS issue based of the error message, however, I'm not sure of the source.
I verified this with a docker build of my own and from other pods in my cluster. Seems to be a Ubuntu related issue, mentioned here: https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/#known-issues. Trying to fix that now.
 
> I will follow up with an issue and report back with a debug image you can use to shell into.
@desimone: For me this is not necessary atm. Maybe it's a better idea to advice people to use the dnsutils pod example I came across yesterday. If you want I can look it up again.

When I find the solution I'll post it here, but this is no longer a Pomerium issue anymore ;-)

**Comment by ReneElstgeest** [comment]
Stuffs working again! So this issue can definitely be closed.

Thnx for the support @desimone

**Comment by desimone** [comment]
@ReneElstgeest Glad to hear. What ended up being the problem for future travelers?

**Comment by ReneElstgeest** [comment]
Well, I found out with microk8s inspect that kubelet was failing. I'm also using ufw which has two known issues with microk8s. I used the fixes at the page I mentioned before:

> Seems to be a Ubuntu related issue, mentioned here: https://microk8s.io/docs/troubleshooting#common-issues. Trying to fix that now.

Then I used `microk8s stop` and `microk8s start` to make sure kubelet was starting properly. Finally I reinstalled my original pomerium helm chart and it magically worked again :-).
